bolt/test/binary-analysis/AArch64/gs-pauth-calls.s

```diff
@@ -0,0 +1,676 @@
+// RUN: %clang %cflags -march=armv8.3-a %s -o %t.exe
+// RUN: llvm-bolt-binary-analysis --scanners=pacret %t.exe 2>&1 | FileCheck %s
```

Comment:
I'm wondering if the user interface for this should be adapted?

`pac-ret` is a widely deployed hardening scheme, so just verifying the correct application of pac-ret hardening is something that users presumably want to do, without also checking pauthabi hardening guarantees that do not exist in pac-ret?

Maybe the checking of non-protected indirect calls should happen under a different `--scanners` option?

Reply:
Sounds reasonable, added a separate `pauth` option as "everything PAuth-related". Yet another option could be added for signing and authentication oracles later, as they should be irrelevant if target CPU is known to implement `FEAT_FPAC`.

Reply:
Are there any future-compatibility concerns with the `--scanners` option? If so, I'd suggest that "everything PAuth-related" is potentially ambiguous, for example if we have some data-PAC usage at some point, or if there are some A/B key restrictions. I wonder if `pauth-abi`, `pauthtest` (to match `-mabi`) or `pauth-inst` (since this scanner is fairly general) might be better. That said, perhaps `pauth` will always be a useful catch-all option, and in that case we can extend this later as required.